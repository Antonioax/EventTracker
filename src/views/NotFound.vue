<script setup>
defineProps({
  resource: {
    type: String,
    required: true,
    default: "page"
  }
});
</script>

<template>
  <div class="text-center">
    <h1 class="text-5xl mb-4">Oops!</h1>
    <h3 class="text-md mb-10">The {{ resource }} you're looking for is not here.</h3>
    <RouterLink :to="{ name: 'EventList' }" class="text-2xl">Take me home</RouterLink>
  </div>
</template>
