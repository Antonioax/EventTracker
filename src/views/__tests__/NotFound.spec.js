import { mount } from "@vue/test-utils";
import { beforeEach, describe, expect, it } from "vitest";

import NotFound from "@/views/NotFound.vue";

describe("NotFound.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(NotFound);
  });

  it("should mount component", () => {
    expect(wrapper).toBeTruthy();
  });

  it("should render default text if no resource", () => {
    expect(wrapper.text()).toContain("The page you're looking for is not here.");
  });

  it("should render resource text if resource given as a prop", async () => {
    await wrapper.setProps({ resource: "event" });
    expect(wrapper.text()).toContain("The event you're looking for is not here.");
  });

  it("should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
