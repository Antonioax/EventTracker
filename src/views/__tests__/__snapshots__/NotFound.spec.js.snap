// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound.vue > should match snapshot 1`] = `
"<div class="text-center">
  <h1 class="text-5xl mb-4">Oops!</h1>
  <h3 class="text-md mb-10">The page you're looking for is not here.</h3>
  <routerlink to="[object Object]" class="text-2xl">Take me home</routerlink>
</div>"
`;
